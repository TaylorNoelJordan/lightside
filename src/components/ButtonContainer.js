import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

import './ButtonContainer.scss'
import human from '../images/006-human.svg';
import planet from '../images/007-universe.svg';
import vehicle from '../images/002-star-wars.svg';


const ButtonContainer = ({ selected, animateButtons }) => {
    let btnData = [
        {   link: '/people',
            title: 'People',
            img: human,
            animateButtons: {animateButtons},
            key: 1
        },
        {   link: '/planets',
            title: 'Planets',
            img: planet,
            animateButtons: {animateButtons},
            key: 2
        },
        {   link: '/vehicles',
            title: 'Vehicles',
            img: vehicle,
            animateButtons: {animateButtons},
            key: 3
        }
    ];

    const buttons = btnData.map(type => {
        return <Button 
                animateButtons={animateButtons}
                link={type.link}
                title={type.title}
                img={type.img}
                key={type.key}
                />
    })
    return (
        <nav className={selected ? 'clickedContainer' : 'btnContainer' }>
            { buttons }
        </nav>
    )

}

ButtonContainer.propTypes = {
    animateButtons: PropTypes.func.isRequired
}

export default ButtonContainer;