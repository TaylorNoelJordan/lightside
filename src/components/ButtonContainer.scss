.btnContainer {
    display: flex;
    justify-content: space-around;
    padding: 30px;
    width: 100%;
  
  }
  
  .clickedContainer {
    flex-direction: column;
    transition: 2s;
    width: 30%;
    float: left;
    button {
        height: 200px;
        width: 200px;
    }
  }
  
  .clickedMain {
    display: flex;
  }
  
//   .active {
//     color: #feda4a;
//     height: 50px;
//     width: 50px;
//   }