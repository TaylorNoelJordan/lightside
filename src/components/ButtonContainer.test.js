import React from 'react';
import { shallow } from 'enzyme';
import ButtonContainer from './ButtonContainer';

describe('ButtonContainer', () => {
    let wrapper;
    let mockData;
    let mockAnimateButtons;

    beforeEach(() => {
        wrapper = shallow(<ButtonContainer />)
        mockData = [
            {
                link: '/people',
                title: 'People',
                img: 'img',
                animateButtons:{mockAnimateButtons},
                key: 1
        },
        {
            link: '/planets',
            title: 'Planets',
            img: 'img',
            animateButtons:{mockAnimateButtons},
            key: 2
    }
        ]
    })

    it('should render button data', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})