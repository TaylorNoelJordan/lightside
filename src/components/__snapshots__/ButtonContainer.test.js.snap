// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should render button data 1`] = `
<nav
  className="btnContainer"
>
  <Button
    img="006-human.svg"
    key="1"
    link="/people"
    title="People"
  />
  <Button
    img="007-universe.svg"
    key="2"
    link="/planets"
    title="Planets"
  />
  <Button
    img="002-star-wars.svg"
    key="3"
    link="/vehicles"
    title="Vehicles"
  />
</nav>
`;
